parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -35,31 +38,79 @@ cleanupContainerdDlFiles() {
     rm -rf $CONTAINERD_DOWNLOADS_DIR
 }
 
-installContainerRuntime() {
-    if [ "${NEEDS_CONTAINERD}" == "true" ]; then
-        echo "in installContainerRuntime - KUBERNETES_VERSION = ${KUBERNETES_VERSION}"
-        local containerd_version
-        if [ -f "$MANIFEST_FILEPATH" ]; then
-            containerd_version="$(jq -r .containerd.edge "$MANIFEST_FILEPATH")"
-            if [ "${UBUNTU_RELEASE}" == "18.04" ]; then
-                containerd_version="$(jq -r '.containerd.pinned."1804"' "$MANIFEST_FILEPATH")"
-            fi
-        else
-            echo "WARNING: containerd version not found in manifest, defaulting to hardcoded."
-        fi
+# After the centralized packages changes, the containerd versions are only available in the components.json.
+installContainerdWithComponentsJson() {
+    os=${UBUNTU_OS_NAME}
+    if [[ -z "$UBUNTU_RELEASE" ]]; then
```

Comment:
can't we just check `if [ "${OS}" == "$UBUNTU_OS_NAME" ] `?

Reply:
ah, I guess we don't really have access to that here, since these functions are supposed to be "common", in that they just work for both ubuntu + mariner (the distro-specific ones are in cse_install_ubuntu.sh/cse_install_mariner.sh)

Reply:
You are right. We don't know what os it is yet. So we needed to use the command `UBUNTU_RELEASE=$(lsb_release -r -s)` to identify it.